src/directives/formioElement.js

```diff
@@ -11,9 +11,18 @@ module.exports = [
         element.replaceWith($compile($templateCache.get(scope.template))(scope));
         scope.$emit('formElementRender', element);
       },
-      controller: function() {
-        // This is required for some reason as it will occasionally throw an error without it.
-      }
+      controller: [
+        '$scope',
+        function(
+          $scope
+        ) {
+          $scope.$watchCollection('data.' + $scope.component.key, function(_new, _old) {
```

Comment:
This will probably have some serious performance issues since this is adding a ```watchCollection``` for every component in the system. It may be more performant to add the watch within the formio directive, and then have it pass that down into the components as the data changes. Thoughts?